src/backends/plonky2/circuits/mainpod.rs

```diff
@@ -137,91 +137,211 @@ impl OperationVerifyGate {
+                            i,
+                        ),
+                    )
+                })
+                .collect::<Vec<_>>()
+        };
+
         // The verification may require aux data which needs to be stored in the
         // `OperationVerifyTarget` so that we can set during witness generation.
 
         // For now only support native operations
-        builder.connect(op.op_type[0], one);
-        let native_op = op.op_type[1];
-
-        let mut op_flags = Vec::new();
-        let op_none = builder.constant(F::from_canonical_u64(NativeOperation::None as u64));
-        let is_none = builder.is_equal(native_op, op_none);
-        op_flags.push(is_none);
-        let op_new_entry =
-            builder.constant(F::from_canonical_u64(NativeOperation::NewEntry as u64));
-        let is_new_entry = builder.is_equal(native_op, op_new_entry);
-        op_flags.push(is_new_entry);
-        let op_copy_statement =
-            builder.constant(F::from_canonical_u64(NativeOperation::CopyStatement as u64));
-        let is_copy_statement = builder.is_equal(native_op, op_copy_statement);
-        op_flags.push(is_copy_statement);
-        let op_eq_from_entries = builder.constant(F::from_canonical_u64(
-            NativeOperation::EqualFromEntries as u64,
-        ));
-        let is_eq_from_entries = builder.is_equal(native_op, op_eq_from_entries);
-        op_flags.push(is_eq_from_entries);
-        let op_lt_from_entries =
-            builder.constant(F::from_canonical_u64(NativeOperation::LtFromEntries as u64));
-        let is_lt_from_entries = builder.is_equal(native_op, op_lt_from_entries);
-        op_flags.push(is_lt_from_entries);
-        let op_not_contains_from_entries = builder.constant(F::from_canonical_u64(
-            NativeOperation::NotContainsFromEntries as u64,
-        ));
-        let is_not_contains_from_entries =
-            builder.is_equal(native_op, op_not_contains_from_entries);
-        op_flags.push(is_not_contains_from_entries);
-
-        // One supported operation must be used.  We sum all operation flags and expect the result
-        // to be 1.  Since the flags are boolean and at most one of them is true the sum is
-        // equivalent to the OR.
-        let or_op_flags = op_flags
-            .iter()
-            .map(|b| b.target)
-            .fold(_false.target, |acc, x| builder.add(acc, x));
-        builder.connect(or_op_flags, _true.target);
+        // Op checks to carry out. Each 'eval_X' should be thought of
+        // as 'eval' restricted to the op of type X, where the
+        // returned target is `false` if the input targets lie outside
+        // of the domain.
+        let op_checks = vec![
+            vec![
+                self.eval_none(builder, st, op),
+                self.eval_new_entry(builder, st, op, prev_statements),
+            ],
+            // Skip these if there are no resolved op args
+            if resolved_op_args.len() == 0 {
```

Comment:
This case only would happen if we configure the Pod to have 0 input signed pods and 0 input main pods right? From `prev_statements.len() == 0`
Do you think we'll ever have this configuration?